### PR TITLE
Add IDEActivityLogMajorGroupSection and fix a potential encoding error

### PR DESCRIPTION
**Pull request by gyfelton**
1. Fix Encoding::UndefinedConversionError:  "\xE2" from ASCII-8BIT to UTF-8" during string parsing
2. Fix undefined class IDEActivityLogMajorGroupSection error

### CODE REVIEW

**Comment by gyfelton** [comment]
Tested using a private xcactivitylog and it works fine

File: lib/xcactivitylog/objects.rb [hunk 1]
```diff
@@ -151,10 +151,10 @@ def write_chrome_trace_file(section_type:, to:)
             tid: thread_id,
             ts: section.time_started_recording_usec,
             ph: 'X',
-            name: section.title,
+            name: section.title.dup&.force_encoding('UTF-8'),
```
Comment (segiddins) [inline lib/xcactivitylog/objects.rb:154]: can you add a test for the fixed encoding issue?

Comment (gyfelton) [reply lib/xcactivitylog/objects.rb:154]: the new test case will fail if this is removed, unless u are talking about a unit test just for this?

Comment (segiddins) [reply lib/xcactivitylog/objects.rb:154]: ah if the new test case fails, then that's sufficient

**Review by segiddins — APPROVED** [review]
Looks good once CI goes green

**Pull request merged**